src/vim9compile.c

```diff
@@ -3970,9 +3971,33 @@ obj_constructor_prologue(ufunc_T *ufunc, cctx_T *cctx)
 
 	if (m->ocm_init != NULL)
 	{
-	    char_u *expr = m->ocm_init;
-
-	    if (compile_expr0(&expr, cctx) == FAIL)
+	    char_u  *expr = m->ocm_init;
+	    int	    ret;
+	    int	    check_extends = FALSE;
+
+	    // An expression containing a dot may use the import namespace
+	    // name of the script on the extended base class.
+	    if (vim_strchr(expr, '.') != NULL && cl_extends != NULL
+		    && cl_extends->class_class_function_count_child > 0)
+		check_extends = TRUE;
+
+	    if (check_extends)
+		++emsg_skip;
+	    ret = compile_expr0(&expr, cctx);
+	    if (check_extends)
+		--emsg_skip;
+	    if (ret == FAIL && check_extends)
+	    {
+		sctx_T current_sctx_save = current_sctx;
+
+		expr = m->ocm_init;
+		current_sctx =
+			cl_extends->class_class_functions[0]->uf_script_ctx;
+		//current_sctx.sc_seq = current_sctx_save.sc_seq;
```

Comment:
The commented out line should be removed?

Reply:
Yes, I'll remove it after I get home.

Reply:
Done.
Thanks for reviewing :+1: